File: ELearning_NUM_Lesson_5/5-2-newt/julia/newt.jl
# Newton-Raphson

f(x) = x^x - 100

fp(x) = x^x * (log(x) + 1)

function NewtonRaphson(x0)
    maxIter = 1000
    tol = 1e-8

    iter = 0
    x = x0
    fx = f(x)
    while abs(fx) > tol && iter < maxIter
        x = x  - fx/fp(x)
        fx = f(x)
        iter += 1
    end
    return x
end

NewtonRaphson(1)

File: ELearning_NUM_Lesson_5/5-2-newt/octave/newt.m
% Newton-Raphson

disp("Newton-Raphson")

function y = f(x)
    y = x^x - 100;
end

function y = fp(x)
    y = x^x * (log(x) + 1);
end

function x = NewtonRaphson(x0)
    maxIter = 1000;
    tol = 1e-8;

    iter = 0;
    x = x0;
    fx = f(x);
    while abs(fx) > tol && iter < maxIter
        x = x - fx/fp(x);
        fx = f(x);
        iter = iter + 1;
    end
    x;
end

NewtonRaphson(1)

